Sources/PubNub/Subscription/SubscriptionSession.swift

```diff
@@ -163,15 +176,25 @@ class SubscriptionSession: EventEmitter, StatusEmitter {
       presenceOnly ? [$0.presenceChannelName] : [$0, $0.presenceChannelName]
     }
     internalUnsubscribe(
-      from: globalChannelSubscriptions.compactMap { channelNamesToUnsubscribe.contains($0.key) ? $0.value : nil },
-      and: globalGroupSubscriptions.compactMap { groupNamesToUnsubscribe.contains($0.key) ? $0.value : nil },
+      from: globalChannelSubscriptions.lockedRead { $0 }.compactMap {
+        channelNamesToUnsubscribe.contains($0.key) ? $0.value : nil
+      },
+      and: globalGroupSubscriptions.lockedRead { $0 }.compactMap {
+        groupNamesToUnsubscribe.contains($0.key) ? $0.value : nil
+      },
```

Comment:
`compactMap ` after closure, close will operate on the reference to the original `elements` (they did not mutate in `lockedRead` closure and COW not applied) but outside of lock. The following change should be more thread-safe:
```suggestion
      from: globalChannelSubscriptions.lockedRead { $0.compactMap {
        channelNamesToUnsubscribe.contains($0.key) ? $0.value : nil
      } },
      and: globalGroupSubscriptions.lockedRead { $0.compactMap {
        groupNamesToUnsubscribe.contains($0.key) ? $0.value : nil
      } },
```